controller/kongplugininstallation/image/image.go

```diff
@@ -23,8 +24,9 @@ import (
 	"github.com/kong/gateway-operator/modules/manager/metadata"
 )
 
-// FetchPluginContent fetches the content of the plugin from the image URL. When authentication is not needed pass nil.
-func FetchPluginContent(ctx context.Context, imageURL string, credentialsStore credentials.Store) ([]byte, error) {
+// FetchPlugin fetches the content of the plugin from the image URL. When authentication is not needed pass nil.
+// It returns a map with two entries - handler.lua and schema.lua, which are the content of the plugin.
+func FetchPlugin(ctx context.Context, imageURL string, credentialsStore credentials.Store) (map[string]string, error) {
```

Comment:
Given that we know what the required files are upfront, why not define a struct instead of the `map[string]string`? E.g.

```go
type PluginFiles struct {
  Schema []byte
  Handler []byte
}
```

Reply:
My assumption was that `FetchPlugin` is able to ensure that the struct it would return is valid, so downstream doesn't have to do that the second time - it'd only use the struct's fields and map them to the file names in the CM's config.

I'm fine with keeping using `map[string]string`, but how about at least a type definition for it and a godoc to make it clear what maps to what exactly? :) 

```go
// PluginFiles maps a plugin's file names to their content.
// It's expected that each plugin consists of `schema.lua` and `handler.lua` files.
type PluginFiles map[string]string
```